### PR TITLE
feat: build react/vue/elements icons

### PR DESCRIPTION
**Pull request by AnnaRybkina**
Background: https://nmp-jira.atlassian.net/jira/software/projects/WARP/boards/15?selectedIssue=WARP-224

- [x] build icons for react
- [x] build icons for vue
- [x] build icons for elements
- [x] add eik import-map

Publishing to NPM and EIK we will be handled in a separate PR.

### CODE REVIEW

**Review by magnuh — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -2,46 +2,77 @@
...
+    "./elements": "./elements/index.js",
+    "./elements/*": "./elements/*.js",
+    "./package.json": "./package.json"
   },
   "scripts": {
     "import": "node ./scripts/import.js",
     "build": "node ./scripts/build.js",
-    "render": "node preview/render.js",
     "dev": "vite",
     "vite:build": "vite build",
```
Comment (BalbinaK) [inline package.json:26]: Looks like `vite build` is only used in `"preview"`, so maybe it's not necessary to keep it as a separate script?

File: package.json [hunk 2]
```diff
@@ -2,46 +2,77 @@
...
-    "preview": "pnpm run build && vite build && vite preview"
+    "preview": "pnpm run build && vite build && vite preview",
+    "build:release": "rimraf dist && pnpm run build && pnpm build:vue && pnpm build:react && pnpm build:elements",
```
Comment (BalbinaK) [inline package.json:28]:
```suggestion
    "build:release": "rimraf dist && pnpm build && pnpm build:vue && pnpm build:react && pnpm build:elements",
```

File: package.json [hunk 3]
```diff
@@ -2,46 +2,77 @@
...
+    "./package.json": "./package.json"
   },
   "scripts": {
     "import": "node ./scripts/import.js",
     "build": "node ./scripts/build.js",
-    "render": "node preview/render.js",
     "dev": "vite",
     "vite:build": "vite build",
-    "preview": "pnpm run build && vite build && vite preview"
+    "preview": "pnpm run build && vite build && vite preview",
```
Comment (BalbinaK) [inline package.json:27]: Just to align with other examples :) 
```suggestion
    "preview": "pnpm build && vite build && vite preview",
```

File: package.json [hunk 4]
```diff
@@ -2,46 +2,77 @@
...
+      "https://assets.finn.no/map/react/v2",
+      "https://assets.finn.no/map/vue/v1",
+      "https://assets.finn.no/map/custom-elements/v2"
+    ]
+  },
+  "config": {
+    "commitizen": {
+      "path": "./node_modules/cz-conventional-changelog"
+    }
   }
```
Comment (BalbinaK) [inline package.json:77]: With that it would be nice to add `"commit": "cz"` to `scripts`.

File: package.json [hunk 5]
```diff
@@ -2,46 +2,77 @@
...
+    },
+    "import-map": [
+      "https://assets.finn.no/map/react/v2",
+      "https://assets.finn.no/map/vue/v1",
+      "https://assets.finn.no/map/custom-elements/v2"
+    ]
+  },
+  "config": {
+    "commitizen": {
+      "path": "./node_modules/cz-conventional-changelog"
```
Comment (BalbinaK) [inline package.json:75]: This requires installing `cz-conventional-changelog` dependency.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          version: 8
+          run_install: true
+      - name: Build
+        run: pnpm build:release
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: npx semantic-release
```
Comment (BalbinaK) [inline .github/workflows/release.yml:31]: We'll need to install semantic-release and its plugins to complete this job, like specified in our [Automate releases guide](https://github.com/warp-ds/release-playground/wiki/Automate-releases-with-semantic-release-and-Eik).

File: package.json [hunk 7]
```diff
@@ -2,46 +2,77 @@
...
+    "files": {
+      "react": "./dist/react",
+      "vue": "./dist/vue",
+      "elements": "./dist/elements"
+    },
+    "import-map": [
+      "https://assets.finn.no/map/react/v2",
+      "https://assets.finn.no/map/vue/v1",
+      "https://assets.finn.no/map/custom-elements/v2"
+    ]
```
Comment (BalbinaK) [inline package.json:71]: Ah! That's what was missing for the imports to be mapped to eik urls in the built files! 👏

**Review by BalbinaK — COMMENTED** [review]
I have some comments that made me think that we should maybe handle publishing/automated releases in a separate PR. This PR is already a bit too big imo ☺️ 

On that note, I think we should aim at having smaller PRs in the future. Especially when we squash-merge PRs that are not going into`main`. The commit messages then don't cover all that has been changed well enough in my opinion. It's also easier to get good reviews when there's less to look through in the changed files 😊

**Comment by BalbinaK** [comment]
I opened a PR just for updating dependencies: https://github.com/warp-ds/icons/pull/6 so we can include fewer changes in this PR.

File: package.json (hunk 3 above)
Comment (BalbinaK) [reply package.json]: Fixed in [da0b941](https://github.com/warp-ds/icons/pull/4/commits/da0b941617db26a10159664a6b7a588aa76c23bf). Here I also renamed the command that shows a preview of all icons to `pnpm dev`. This way all our repos have a similar command to see the output of that package locally. Tested that the output is the same when using `pnpm build && vite` as in `pnpm build && vite build && vite preview`.

File: package.json (hunk 1 above)
Comment (BalbinaK) [reply package.json]: [da0b941](https://github.com/warp-ds/icons/pull/4/commits/da0b941617db26a10159664a6b7a588aa76c23bf)

File: .github/workflows/release.yml (hunk 6 above)
Comment (BalbinaK) [reply .github/workflows/release.yml]: Moved changes related to automated releases to https://github.com/warp-ds/icons/pull/7

File: package.json (hunk 5 above)
Comment (BalbinaK) [reply package.json]: Moved changes related to automated releases to https://github.com/warp-ds/icons/pull/7

**Comment by BalbinaK** [comment]
Sorry for the mess. I wanted to move the logic for automating releases out to make the changes easier to review. There's a draft PR for handling automated releases here: https://github.com/warp-ds/icons/pull/7.

**Review by siljec — APPROVED** [review]
LGTM 🎉

**Pull request merged**